Debugging Controls

ElMarchand opened the pull request:
Allow users to interact with the game even without a VR headset for development purposes.

![Controls (0)](https://github.com/med-material/Whack_A_Mole_VR/assets/91123329/eb6bc3a6-36c3-460f-9293-cbd26a86d74f)

Current controls to play the game:
- WASD/Arrow keys to move in lines (works diagonally)
- When holding the Left Alt key, moving your mouse will move the cursor at the same time
 - The cursor speed depends on the user's mouse sensitivity, so it might be very fast.

Fixes [ Add Debugging Controls (keyboard/mouse) when no VR is present #278 ](https://github.com/med-material/Whack_A_Mole_VR/issues/278) 

New pull request due to an error on my last one, which I deleted since.

bastianilso commented on Assets/Scripts/Pointers/BasicPointer.cs:
```diff
@@ -25,11 +26,53 @@ public class BasicPointer : Pointer
     private delegate void Del();
     private string hover = "";
-
+    public Vector3 CalculateDirection()
+    {
+        Vector3 direction = Vector3.zero;
+        if (Input.GetKey(KeyCode.UpArrow) || Input.GetKey(KeyCode.W))
+        {
+            direction.y += 0.25f;
+        }
+        if (Input.GetKey(KeyCode.LeftArrow) || Input.GetKey(KeyCode.A))
+        {
+            direction.x -= 0.25f;
+        }
+        if (Input.GetKey(KeyCode.DownArrow) || Input.GetKey(KeyCode.S))
+        {
+            direction.y -= 0.25f;
+        }
+        if (Input.GetKey(KeyCode.RightArrow) || Input.GetKey(KeyCode.D))
+        {
+            direction.x += 0.25f;
+        }
+        return direction.normalized;
+    }
```
The CalculateDirection function is not called anywhere in this code.

ElMarchand replied:
CalculateDirection is only called once to assign different keys and speeds to movement axes. The "direction" vector is used when moving the controller with the arrow keys.

bastianilso commented on Assets/Scripts/Pointers/BasicPointer.cs:
```diff
@@ -25,11 +26,53 @@ public class BasicPointer : Pointer
+        return direction.normalized;
+    }
+    [SerializeField]
+    private GameObject Cube;
+    
     // Function called on VR update, since it can be faster/not synchronous to Update() function. Makes the Pointer slightly more reactive.
-    public override void PositionUpdated()
+    public void Update()
     {
         if (!active) return;
+        if (active)
+        {
+            float v = Input.GetAxisRaw("Vertical");
+            float h = Input.GetAxisRaw("Horizontal");
+            float mouseSpeed = 1;
+
+            if (Input.GetKey(KeyCode.LeftAlt))
+            {
+                float h1 = mouseSpeed * Input.GetAxis("Mouse X");
+                float v1 = mouseSpeed * Input.GetAxis("Mouse Y");
+
+                Cube.transform.Translate(h1, v1, 0);
+            }
+            else
+            {
+                Vector3 direction = new Vector3(h, v, 0f).normalized;
+                Cube.transform.Translate(direction * 1 * Time.deltaTime);
+            }
+        };
```
this code seems to be moving a "Cube" object instead of moving the actual controller?

bastianilso reviewed (changes requested):
Hi Elric, much cleaner now (only 2 files changed by the commit). However, I'm not quiet sure how its supposed to work (are we sure this is the right code?).

Commit:
- Please edit your commit title and message to adhere to [the commit message guidelines I sent you](https://chris.beams.io/posts/git-commit/).

Mainscene:
- Since mainscene is difficult to inspect, I would mainly request you to make a mainscene file where only the changes you intend to make are present. You can do this by downloading the mainscene file from master, applying your changes (adding box if you wish), testing the changes, and saving.

BasicPointer.cs:
- It looks like currently the PositionUpdated() function is being removed from BasicPointer.cs and replaced with just Update(). Please keep PositionUpdated() as it originally was - using PositionUpdated is important for updating VR.

ElMarchand commented:
@bastianilso I pushed a new version, here are the changes :

**BasicPointer.cs**
- Moved all the functions that allow the pointer to work in a new function named "PointerControl", called in VR mode and Debug mode.
- Debug Controls are active only while the LeftControl key is held down, therefore VR and Debug shouldn't be fighting for the pointer.
- Removed relations to the Cube Object used to move the controllers.

**MainScene.unity**
- Removed the Cube Object from the scene.

My second push today was to correct a part that I forgot to delete in the previous commit.

bastianilso commented:
@ElMarchand thanks for pushing the new changes.  Currently the branch contains 3 commits, can you squash them to 1 commit? Thanks.

bastianilso reviewed (changes requested):
@ElMarchand your commit includes some changes to GazeLogger that shouldn't be necessary. Also, if we can make this commit without changes to the MainScene, that would be great too. Thanks! 👍

bastianilso reviewed (changes requested):
Better - what about the mainscene changes are they still necessary?

Also, please correct the commit title and message to reflect what the work we are trying to merge does.
